### ISSUE TITLE
Overlapping instances in fine-tuning SAM

### ISSUE THREAD

**Issue by Ashutosh1995** [bnsreenu/python_for_microscopists#82]
Hello @bnsreenu 

Thanks for providing the repo for fine-tuning the SAM model on the custom dataset. My question is that in case where the instances of the object are overlapping then in that case, how can I change the code to include different boxes for different objects?

Kindly provide your suggestion?
![temp_binary](https://github.com/bnsreenu/python_for_microscopists/assets/11408850/9e0a6f48-d193-4a44-b17d-9be198cbaa3e)

**Comment by RoyAn2386** [comment]
Hi man,

I think you should prepare your label as instance mask (random color for each object), then use finding contour to get different boxes. T.T

**Comment by Ashutosh1995** [comment]
Thanks for your reply! You mean to say in this mask, assign each object with different color as in different ID, apply patchify. In that 256x256 instance, see for which instance is coming and get boxes using contours?

Also I saw in the notebook that you were using ViT-B, doesn't that internally resizes the image to be 1024x1024? If that's the case, what GPU are you using to finetune this model?
